node/nodeInternalBlocks.go

```diff
@@ -0,0 +1,124 @@
+	if n.processComponents.ShardCoordinator().SelfId() != core.MetachainShardId {
+		return nil, ErrMetachainOnlyEndpoint
+	}
+
+	decodedHash, err := hex.DecodeString(hash)
+	if err != nil {
+		return nil, err
+	}
+
+	apiBlockProcessor, err := n.createRawBlockProcessor()
+	if err != nil {
+		return nil, err
+	}
+
+	return apiBlockProcessor.GetInternalMetaBlockByHash(format, decodedHash)
+}
+
+// GetInternalMetaBlockByNonce wil return a meta block by nonce
+func (n *Node) GetInternalMetaBlockByNonce(format common.OutportFormat, nonce uint64) (interface{}, error) {
+	if n.processComponents.ShardCoordinator().SelfId() != core.MetachainShardId {
+		return nil, ErrMetachainOnlyEndpoint
+	}
+
+	apiBlockProcessor, err := n.createRawBlockProcessor()
+	if err != nil {
+		return nil, err
+	}
+
+	return apiBlockProcessor.GetInternalMetaBlockByNonce(format, nonce)
+}
+
+// GetInternalMetaBlockByRound wil return a meta block by round
+func (n *Node) GetInternalMetaBlockByRound(format common.OutportFormat, round uint64) (interface{}, error) {
+	if n.processComponents.ShardCoordinator().SelfId() != core.MetachainShardId {
+		return nil, ErrMetachainOnlyEndpoint
+	}
+
+	apiBlockProcessor, err := n.createRawBlockProcessor()
+	if err != nil {
+		return nil, err
+	}
+
+	return apiBlockProcessor.GetInternalMetaBlockByRound(format, round)
+}
+
+// GetInternalShardBlockByHash wil return a shard block by hash
+func (n *Node) GetInternalShardBlockByHash(format common.OutportFormat, hash string) (interface{}, error) {
+	if n.processComponents.ShardCoordinator().SelfId() == core.MetachainShardId {
+		return nil, ErrShardchainOnlyEndpoint
+	}
+
+	decodedHash, err := hex.DecodeString(hash)
+	if err != nil {
+		return nil, err
+	}
+
+	apiBlockProcessor, err := n.createRawBlockProcessor()
+	if err != nil {
+		return nil, err
+	}
+
+	return apiBlockProcessor.GetInternalShardBlockByHash(format, decodedHash)
+}
+
+// GetInternalShardBlockByNonce wil return a shard block by nonce
+func (n *Node) GetInternalShardBlockByNonce(format common.OutportFormat, nonce uint64) (interface{}, error) {
+	if n.processComponents.ShardCoordinator().SelfId() == core.MetachainShardId {
+		return nil, ErrShardchainOnlyEndpoint
+	}
+
+	apiBlockProcessor, err := n.createRawBlockProcessor()
+	if err != nil {
+		return nil, err
+	}
+
+	return apiBlockProcessor.GetInternalShardBlockByNonce(format, nonce)
+}
+
+// GetInternalShardBlockByRound wil return a shard block by round
+func (n *Node) GetInternalShardBlockByRound(format common.OutportFormat, round uint64) (interface{}, error) {
+	if n.processComponents.ShardCoordinator().SelfId() == core.MetachainShardId {
+		return nil, ErrShardchainOnlyEndpoint
+	}
+
+	apiBlockProcessor, err := n.createRawBlockProcessor()
+	if err != nil {
+		return nil, err
+	}
+
+	return apiBlockProcessor.GetInternalShardBlockByRound(format, round)
+}
+
+func (n *Node) createRawBlockProcessor() (blockAPI.APIRawBlockHandler, error) {
```

Comment:
why do you need to create it each time? it could have been a member of the `Node` struct

Reply:
I just followed the same pattern from APIBlockHandler, will refactor to be more efficient :+1:

Reply:
refactored to be a member of Node struct